backend/src/controllers/products.controller.ts

```diff
@@ -127,6 +128,7 @@ productController.put(
 productController.use(
   '/:productId/transactions',
   verifyToken,
+  checkProductAuthorizationInverted,
```

Comment:
nice.... looks like you understand the concept of middlewares. :wink: 


We have to be careful, that we're not overusing middlewares for database relevant checks. Otherwise we end up getting the product from database multiple times (high cost?) for quick checks that could be done in the end when we need to product anyway (another db query)

This is just an FYI and no changes are necessary